cluster-autoscaler/core/scaledown/actuation/update_latency_tracker.go

```diff
@@ -0,0 +1,79 @@
+package actuation
+
+import (
+	"time"
+
+	"k8s.io/autoscaler/cluster-autoscaler/utils/kubernetes"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/taints"
+	"k8s.io/klog/v2"
+)
+
+type UpdateLatencyTracker struct {
+	startTimestamp  map[string]time.Time
+	finishTimestamp map[string]time.Time
+	nodeLister      kubernetes.NodeLister
+}
+
+func NewUpdateLatencyTracker(startTimestamp map[string]time.Time, nodeLister kubernetes.NodeLister) *UpdateLatencyTracker {
+	return &UpdateLatencyTracker{
+		startTimestamp:  startTimestamp,
+		finishTimestamp: map[string]time.Time{},
+		nodeLister:      nodeLister,
+	}
+}
+
+func (u *UpdateLatencyTracker) Start(stopChan <-chan struct{}, resultChan chan int64) {
+	for {
+		select {
+		case <-stopChan:
+			var remainingNodes []string
+			for node, _ := range u.startTimestamp {
+				remainingNodes = append(remainingNodes, node)
+			}
+			for remainingNodes != nil {
```

Comment:
This is a potentially infinite loop if something goes wrong. There has to be a timeout. I'd actually consider passing a context instead of stopChan here. When this is taking too long, the context can time out. When there's an error, the context can be cancelled.

Reply:
Added a timeout